examples/Boost-python/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2013, Ruslan Baratov
+# Modified work: Copyright (c) 2018, Gregory Kramida
+# All rights reserved.
+
+cmake_minimum_required(VERSION 3.2)
+
+# Emulate HunterGate:
+# * https://github.com/hunter-packages/gate
+set(TESTING_CONFIG_OPT FILEPATH "${CMAKE_CURRENT_LIST_DIR}/config.cmake")
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_LIST_DIR}/toolchain.cmake")
```

Comment:
In general it should be a cache variable so user will have an ability to customize this part.

Reply:
In this example toolchain should be removed because toolchains specified by CI scripts.

Reply:
@ruslo , you asked me to use toolchains in order to get rid of CMAKE_POSITION_INDEPENDENT_CODE usage in the example. I couldn't really find a minimal example or clear-cut instructions on what needs to be done to use a toolchain. The "toolchain.cmake" variant was the best I could come up with after reading [the documentation](https://docs.hunter.sh/en/latest/overview/customization/toolchain-id.html#simple-toolchains). I have no idea how to apply whatever is in the ``polly`` codebase or the Android "example" to my code.

If you would be so kind, please provide more specific information on what exactly is it that you want me to do in order to add set(CMAKE_POSITION_INDEPENDENT_CODE TRUE) wherever it needs to be added.

As to the branches for ```find_package(Python...```, хозяин - барин, как говориться, and I'll remove the branch for the newer versions of CMake, but "Deprecated" means "can, but should not, be used, and will very likely be removed altogether in some future release". You'll have to fix it then when it breaks in the future.

Reply:
Also, which variable to you want as a 'CACHE' variable?

Reply:
@ruslo, the reason for the -fPIC option, if I am not mistaken, is that C/C++ python extension libraries are always built as shared libraries which could be accessed by multiple python processes at the same time. Compiling dependent libraries without the -fPIC option can therefore break user code.

Reply:
> So the right way to test with PIC in Hunter is to add such toolchain to testing table, e.g. use `gcc-7-cxx17-pic` instead of `gcc-7-cxx17` here:
> 
> * https://github.com/ingenue/hunter/blob/d878efc6fe0cab199d3a3e828a3aae8000c836eb/.travis.yml#L41-L44

Thank you for the detailed explanation, I think now I understand the situation.

1) Do we want to keep the toolchain stuff (with cache option) in the example, in spite of it being overridden?
2) Do we want to add another entry into the matrix with gcc-7-cxx17-pic in .travis.yml of https://github.com/ingenue/hunter? Not remove the existing gcc-7-cxx17 one, but add a new one with gcc-7-cxx17-pic?

Reply:
> If you want Boost libraries to be shared you can use [HUNTER_BUILD_SHARED_LIBS=ON](https://docs.hunter.sh/en/latest/reference/user-variables.html#hunter-build-shared-libs), example:
> 
> * https://github.com/ruslo/hunter/blob/v0.23.74/examples/tvm/CMakeLists.txt#L6

I understand, I don't want them to be shared. That creates problems for Windows users.

Reply:
> Do we want to keep the toolchain stuff (with cache option) in the example, in spite of it being overridden?

No, it doesn't fit the system anyway. Please remove toolchain related code.

> Do we want to add another entry into the matrix with gcc-7-cxx17-pic in .travis.yml of https://github.com/ingenue/hunter? Not remove the existing gcc-7-cxx17 one, but add a new one with gcc-7-cxx17-pic?

It will be new ` examples/Boost-python` entry with `gcc-7-cxx17-pic`.

> I understand, I don't want them to be shared. That creates problems for Windows users, where Boost isn't in the system path by default.

Hunter CI do not run tests, we only build (configure, compile, link), so it should not be a problem.

Reply:
In this case, I'll try to build everything as shared libraries as in the example and forego changing the Travis CI script.

Reply:
Seems to be working.